tests/helpers.nim

```diff
@@ -8,41 +8,29 @@ import unittest2
 export unittest2
 
 const
-  StreamTransportTrackerName = "stream.transport"
-  StreamServerTrackerName = "stream.server"
   DgramTransportTrackerName = "datagram.transport"
+  UdpTransportTrackerName = "webrtc.udp.transport"
+  StunTransportTrackerName = "webrtc.stun.transport"
+  StunConnectionTrackerName = "webrtc.stun.connection"
+
 
   trackerNames = [
-    StreamTransportTrackerName,
-    StreamServerTrackerName,
     DgramTransportTrackerName,
+    UdpTransportTrackerName,
+    StunTransportTrackerName,
+    StunConnectionTrackerName,
   ]
 
-template asyncTest*(name: string, body: untyped): untyped =
-  test name:
-    waitFor((proc () {.async, gcsafe.} = body)())
-
-iterator testTrackers*(extras: openArray[string] = []): TrackerBase =
-  for name in trackerNames:
-    let t = getTracker(name)
-    if not isNil(t): yield t
-  for name in extras:
-    let t = getTracker(name)
-    if not isNil(t): yield t
-
-template checkTracker*(name: string) =
-  var tracker = getTracker(name)
-  if tracker.isLeaked():
-    checkpoint tracker.dump()
-    fail()
-
 template checkTrackers*() =
-  for tracker in testTrackers():
-    if tracker.isLeaked():
-      checkpoint tracker.dump()
-      fail()
+  var didTrackersLeaked = false
```

Comment:
We already have this in nim-libp2p, what do you think about moving the code to a common project and avoid the duplication? How are other projects handling this currently?

It's not clear in the nim-libp2p why `GC_fullCollect()` was added there. Without a clear and evidence based explanation of why this is needed, we shouldn't add it.

Reply:
> We already have this in nim-libp2p, what do you think about moving checkTrackers to a common project and avoid the duplication? How are other projects handling this currently?

I mean, it can be moved to chronos, I guess.

> It's not clear in nim-libp2p why GC_fullCollect() was added there. Without a clear and evidence based explanation of why this is needed, we shouldn't add it.

From what I can gather, GC_fullCollect() force a full clean of the garbage collector. In my opinion, it can avoid some side effects even if I admit that it's a bit overkill.

Reply:
> I mean, it can be moved to chronos, I guess.

Could you please take care of this?

> From what I can gather, GC_fullCollect() force a full clean of the garbage collector. In my opinion, it can avoid some side effects even if I admit that it's a bit overkill.

What kind of side effects? I'd remove it and see if it causes problems - which would probably be weird. If it does, we either fix the problem - if it's not too time-consuming - or add it back with a clear explanation of why it is necessary.

Reply:
>  I mean, it can be moved to chronos, I guess.

Is it already here? https://github.com/status-im/nim-chronos/blob/master/chronos/unittest2/asynctests.nim

Reply:
> Is it already here? https://github.com/status-im/nim-chronos/blob/master/chronos/unittest2/asynctests.nim

Hum... I tried it and there is a problem. As checkLeaks is a proc, when there's a leak, it shows that everything is right then ends with an error :
```
Stun message encoding/decoding .. (0.00s)
Stun checkForError .. (0.00s)
[Summary] 4 tests run (0.00s): 4 OK, 0 FAILED, 0 SKIPPED
Error: execution of an external program failed: '/tmp/foo '
```

Reply:
Maybe related to https://github.com/status-im/nim-unittest2/issues/43 ?

Reply:
This was an old issue from unittest2 https://github.com/status-im/nim-unittest2/issue/43
I found a roundabout and proposed a PR to solve this issue https://github.com/status-im/nim-unittest2/pull/44

Meanwhile, I just removed helpers.nim and use `proc checkLeaks()` defined in chronos.